### ISSUE TITLE
loadData replaces null with empty string on hsqldb when column not explicitly configured

### ISSUE THREAD

**Issue by hstiekema** [liquibase/liquibase#1159]
## Environment
Windows 10

**Liquibase Version**: 3.9.0 

**Liquibase Integration & Version**: 
SpringLiquibase

**Liquibase Extension(s) & Version**: n/a

**Database Vendor & Version**: 
hsqldb 2.5.0

**Operating System Type & Version**: Windows 10

## Description
loadData replaces null with empty string when the nullable column is not specifically defined. It works as expected when the nullable column is defined or when no column is defined at all.

## Steps To Reproduce
Using liquibase.integration.spring.SpringLiquibase to load my changes

master changelog file:

```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<databaseChangeLog
        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.9.xsd">

    <changeSet id="liquibase-bug-001" author="me">
        <createTable tableName="FOREIGN_KEY_TEST">
            <column name="ID" type="VARCHAR(32 CHAR)">
                <constraints primaryKeyName="PK_TEST" primaryKey="true"/>
            </column>
            <column name="PARENT_FK" type="VARCHAR(32 CHAR)"/>
        </createTable>
    </changeSet>

    <changeSet id="liquibase-bug-002" author="me">
        <loadData tableName="FOREIGN_KEY_TEST" file="testdata.csv" relativeToChangelogFile="true">
            <column name="ID" type="STRING"/>
        </loadData>
    </changeSet>

    <changeSet id="liquibase-bug-003" author="me">
        <addForeignKeyConstraint baseTableName="FOREIGN_KEY_TEST"
                                 baseColumnNames="PARENT_FK"
                                 constraintName="TEST_FK_PARENT"
                                 referencedTableName="FOREIGN_KEY_TEST" referencedColumnNames="ID"/>
    </changeSet>
</databaseChangeLog>
```

Contents of testdata.csv:

```
"ID","PARENT_FK"
"ROOT",null
"CHILD","ROOT"
```

## Actual Behavior
org.hsqldb.HsqlException: integrity constraint violation: foreign key no parent; TEST_FK_PARENT table: FOREIGN_KEY_TEST value: 

When inspecting the actual database, it has loaded the `ROOT, null` record as `ROOT,''`

## Expected/Desired Behavior
The `ROOT,null` record should be loaded as `ROOT,null`

## Additional Context
Changing the configuration of changeset liquibase-bug-002 to either of the following prevents the problem:

```
<changeSet id="liquibase-bug-002" author="me">
        <loadData tableName="FOREIGN_KEY_TEST" file="testdata.csv" relativeToChangelogFile="true">
            <column name="ID" type="STRING"/>
            <column name="PARENT_FK" type="STRING"/>
        </loadData>
    </changeSet>
```

or 

```
<changeSet id="liquibase-bug-002" author="me">
        <loadData tableName="FOREIGN_KEY_TEST" file="testdata.csv" relativeToChangelogFile="true"/>
    </changeSet>
```

**Comment by molivasdat** [comment]
Hi @hstiekema, thanks for writing up this issue and the workaround.

**Comment by pkaloc** [comment]
linux/amd64

Liquibase Version: 3.6.3

Liquibase Integration & Version:
SpringLiquibase

Liquibase Extension(s) & Version: n/a

Database Vendor & Version:
postgres 12.1 (Docker image: postgres:12.1)

Operating System Type & Version: linux/amd64(Docker image: maven:3.6.3-amazoncorretto-11)

Hi there. I am running into the same issue (ie. loadData replaces null with empty string when the nullable column is not specifically defined) with PostgreSQL.

**Comment by FBurguer** [comment]
Looks like it go fixed for liquibase `4.16.1`. You can always reach out if you still have this issue. Thanks!
